### PR TITLE
Add functions for html conversion and notebook stripping

### PR DESCRIPTION
**Pull request by antoine-daigneault-demers**

### CODE REVIEW

File: src/notebook_utils/automation.py [hunk 1]
```diff
@@ -1,20 +1,96 @@
...
 
+def notebook_to_html_cli():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-file",
+        type=pathlib.Path,
+        help="Notebook to convert to html."
+    )
+    parser.add_argument(
+        "-doc",
```
Comment (fmigneault) [inline src/notebook_utils/automation.py]: `-d,--directory` or `-o,--output` would be more appropriate.
This output does not seem related to documentation.

File: src/notebook_utils/automation.py [hunk 2]
```diff
@@ -1,20 +1,96 @@
...
+        type=pathlib.Path,
+        help="Directory where to recursively look for notebooks. Defaults to the current directory.",
+        default=".",
+    )
+    args = parser.parse_args()
+
+    for notebook_file in args.dir.rglob("*.ipynb"):
+        # Ignores ipynb checkpoints
+        if not is_ipynb_checkpoint(notebook_file):
+            strip_notebook(notebook_file)
```
Comment (fmigneault) [inline src/notebook_utils/automation.py]: If any notebook fails, it returns with no error status. Process is incomplete and reported as all success.

File: src/notebook_utils/automation.py [hunk 3]
```diff
@@ -1,20 +1,96 @@
...
+        type=pathlib.Path,
+        help="Notebook to strip."
+    )
+    args = parser.parse_args()
+
+    if args.file is None or args.file.suffix != ".ipynb":
+        _logger.warn("You must specify a valid notebook (.ipynb file) to strip.")
+        exit()
+
+    strip_notebook(args.file)
```
Comment (fmigneault) [inline src/notebook_utils/automation.py]: Failure returns status without error.

File: src/notebook_utils/automation.py [hunk 4]
```diff
@@ -1,20 +1,96 @@
...
+        help="Directory where the html file will be saved. Defaults to the current directory.",
+        default=".",
+    )
+    args = parser.parse_args()
+
+    if args.file is None or args.file.suffix != ".ipynb":
+        _logger.warn("You must specify a valid notebook (.ipynb file) to convert.")
+        exit()
+
+    notebook_to_html(args.file, args.doc)
```
Comment (fmigneault) [inline src/notebook_utils/automation.py]: No error

File: src/notebook_utils/automation.py [hunk 5]
```diff
@@ -1,20 +1,96 @@
...
+        type=pathlib.Path,
+        help="Directory where the html files will be saved. Defaults to the current directory.",
+        default=".",
+    )
+    args = parser.parse_args()
+
+    for notebook_file in args.dir.rglob("*.ipynb"):
+        # Ignores ipynb checkpoints
+        if not is_ipynb_checkpoint(notebook_file):
+            notebook_to_html(notebook_file, args.doc)
```
Comment (fmigneault) [inline src/notebook_utils/automation.py]: No error

**Review by fmigneault — CHANGES_REQUESTED** [review]
Any failing execution returns `exit 0` leaving the false impression it succeeded. 
This is even more a problem with loops since operation can be only partially done.

Consider replacing arguments like `-file` to `-f`/`--file` (same for other arguments).
The single `-` is usually employed as shortcut for `--` long names.

**Review by fmigneault — CHANGES_REQUESTED** [review]
Failing operations do not return the non zero error code.
When looping files, process could look like it succeeded although it worked only for part of the files.

The arguments with single `-` should be replaced by `--` long notation. 
The short notation (e.g.: `-f` for `--file`) can also be added.

**Comment by antoine-daigneault-demers** [comment]
Should I return an error message in addition to an exit code 1 on errors?

RIght now the errors from the commands themselves show up in the console, is it enough?

**Comment by fmigneault** [comment]
> Should I return an error message in addition to an exit code 1 on errors?
> 
> RIght now the errors from the commands themselves show up in the console, is it enough?

A simple message could be printed to help indicate what went wrong, but not the full exception traceback to make the CLI "cleaner". 
The return value should be only the error code. There could be some verbosity flags to print more or less information using for example logger levels (example: https://github.com/crim-ca/weaver/blob/master/weaver/cli.py#L989-L1030).

File: src/notebook_utils/automation.py [hunk 6]
```diff
@@ -1,20 +1,108 @@
...
+        help="Directory where to recursively look for notebooks. Defaults to the current directory.",
+        default=".",
+    )
+    args = parser.parse_args()
+
+    exit_status = []
+    for notebook_file in args.dir.rglob("*.ipynb"):
+        # Ignores ipynb checkpoints
+        if not is_ipynb_checkpoint(notebook_file):
+            exit_status.append(strip_notebook(notebook_file))
```
Comment (fmigneault) [inline src/notebook_utils/automation.py]: To help the user, you should add an error log of which notebook file had a problem.
Same thing for the other CLI loop.

**Review by fmigneault — COMMENTED** [review]
Only check to add some logging details. The rest is fine.

**Comment by antoine-daigneault-demers** [comment]
Thanks for the feedback!

File: src/notebook_utils/automation.py [hunk 7]
```diff
@@ -31,13 +32,26 @@ def notebook_to_html_cli():
...
 
+    if args.verbose:
+        _logger.setLevel(logging.INFO)
+
     if args.file is None or args.file.suffix != ".ipynb":
-        _logger.warn("You must specify a valid notebook (.ipynb file) to convert.")
+        _logger.error("You must specify a valid notebook (.ipynb file) to convert.")
+        sys.exit(1)
+
+    _logger.info(f"Converting {args.file}.")
```
Comment (fmigneault) [inline src/notebook_utils/automation.py]: Logger parameters should use this format to take advantage of skipping string formatting if level is disabled:
```python
_logger.info("Converting %s.", args.file)
```

(other places to edit as well)

**Review by fmigneault — COMMENTED** [review]
Looks fine. Just need to patch the logger calls and good to merge.

**Review by fmigneault — APPROVED** [review]

**Comment by fmigneault** [comment]
Make sure to apply the version tag after merge.

**Pull request merged**